src/matchers/tables.test.lua

```diff
@@ -0,0 +1,501 @@
+			assertMatcher.hasMessage(
+				result,
+				("Expected %s to not contain %s with deep equality"):format(
+					prettyValue(testArray),
+					prettyValue(testItem)
+				)
+			)
+		end)
+	end)
+
+	describe("toMatchObject", function()
+		describe("should pass", function()
+			test(
+				"when the received object matches the expected object",
+				function()
+					-- Given
+					local testObject = {
+						a = 10,
+						b = "eleven",
+						c = "quux",
+					}
+
+					local testMatchObject = {
+						a = 10,
+						b = "eleven",
+					}
+
+					-- When
+					local result = matchers.toMatchObject(
+						CONTEXT,
+						testObject,
+						testMatchObject
+					)
+
+					-- Then
+					assertMatcher.passed(result)
+				end
+			)
+
+			test(
+				"when the received array of objects matches the expected array of objects",
+				function()
+					-- Given
+					local testObjectArray = {
+						{
+							a = 10,
+							b = "eleven",
+							c = "quux",
+						},
+
+						{
+							foo = 85,
+							bar = "lest",
+						},
+					}
+
+					local testMatchObjectArray = {
+						{
+							a = 10,
+							b = "eleven",
+						},
+
+						{ bar = "lest" },
+					}
+
+					-- When
+					local result = matchers.toMatchObject(
+						CONTEXT,
+						testObjectArray,
+						testMatchObjectArray
+					)
+
+					-- Then
+					assertMatcher.passed(result)
+				end
+			)
+		end)
+
+		describe("should fail", function()
+			test(
+				"when the received object doesn't match the expected object",
+				function()
+					-- Given
+					local testObject = {
+						a = 10,
+						b = "eleven",
+						c = "quux",
+					}
+
+					local testMatchObject = {
+						a = 17,
+						b = "twelve",
+					}
+
+					-- When
+					local result = matchers.toMatchObject(
+						CONTEXT,
+						testObject,
+						testMatchObject
+					)
+
+					-- Then
+					assertMatcher.failed(result)
+					assertMatcher.hasMessage(
+						result,
+						("Expected %s to match %s"):format(
+							prettyValue(testObject),
+							prettyValue(testMatchObject)
+						)
+					)
+				end
+			)
+
+			test(
+				"when the received array of objects doesn't match the expected array of objects",
+				function()
+					-- Given
+					local testObjectArray = {
+						{
+							a = 10,
+							b = "eleven",
+							c = "quux",
+						},
+
+						{
+							foo = 85,
+							bar = "lest",
+						},
+					}
+
+					local testMatchObjectArray = {
+						{
+							a = 10,
+							b = "eleven",
+						},
+
+						{ bar = "lest" },
+
+						{ quux = "fiddle" },
+					}
+
+					-- When
+					local result = matchers.toMatchObject(
+						CONTEXT,
+						testObjectArray,
+						testMatchObjectArray
+					)
+
+					-- Then
+					assertMatcher.failed(result)
+					assertMatcher.hasMessage(
+						result,
+						("Expected %s to match %s"):format(
+							prettyValue(testObjectArray),
+							prettyValue(testMatchObjectArray)
+						)
+					)
+				end
+			)
+
+			test("when the received object is not a table", function()
```

Comment:
If it's not a table, it's not an object by the JS definition of it

```suggestion
			test("when the received value is not a table", function()
```